src/MetaMorpho.sol

```diff
@@ -649,9 +645,18 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         return IMorphoMarketParams(address(MORPHO)).idToMarketParams(id);
     }
 
-    /// @dev Returns the vault's balance the market defined by `marketParams`.
-    function _supplyBalance(MarketParams memory marketParams) internal view returns (uint256) {
-        return MORPHO.expectedSupplyAssets(marketParams, address(this));
+    /// @dev Accrues interest on Morpho Blue and returns the vault's assets & corresponding shares supplied on the
+    /// market defined by `marketParams`, as well as the market's state.
+    /// @dev Assumes that the inputs `marketParams` and `id` match.
+    function _accruedSupplyBalance(MarketParams memory marketParams, Id id)
+        internal
+        returns (uint256 assets, uint256 shares, Market memory market)
+    {
+        MORPHO.accrueInterest(marketParams);
+
+        market = MORPHO.market(id);
```

Comment:
Is it beneficial to have a utility function that, given a `marketID` and `userAddress` returns both the market data and user data?

It would avoid you to perform multiple external calls. But I don't want you to bloat Morpho Blue just because of a one-time thing.

It's anyway something that you should consider evaluating: what utility getters are needed for you (metamorpho, bundlers, URD, and so on) and integrators?

Reply:
I think one can leverage the `MorphoStorageLib` to extract more efficiently data but it would complexify a bit the development. I think the current version is fine tbh